docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+import argparse
+import os
+import re
+import shutil
+from argparse import BooleanOptionalAction
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
```

Comment:
What is the target version of python, in case this will become 3.10+ we can leave that out and write type hints like `int | None` instead of `Optional[int]`

Reply:
Thanks for the advice, I wasn't aware of this new feature of python 3.10. The target currently is 3.9, though.